overcooked_ai_py/agents/benchmarking.py

```diff
@@ -40,45 +40,46 @@ def __init__(self, mdp_params={}, env_params={}, mdp_fn=None, force_compute=Fals
 
         self.force_compute = force_compute
 
-    def evaluate_random_pair(self, num_games=1, all_actions=True, display=False):
+    def evaluate_random_pair(self, num_games=1, all_actions=True, display=False, regen_mdp=True):
         agent_pair = AgentPair(RandomAgent(all_actions=all_actions), RandomAgent(all_actions=all_actions))
-        return self.evaluate_agent_pair(agent_pair, num_games=num_games, display=display)
+        return self.evaluate_agent_pair(agent_pair, num_games=num_games, display=display, regen_mdp=regen_mdp)
 
-    def evaluate_human_model_pair(self, num_games=1, display=False):
-        a0 = GreedyHumanModel(self.mlp)
-        a1 = GreedyHumanModel(self.mlp)
+    def evaluate_human_model_pair(self, num_games=1, display=False, regen_mdp=True):
+        a0 = GreedyHumanModel(self.env.mlp)
+        a1 = GreedyHumanModel(self.env.mlp)
         agent_pair = AgentPair(a0, a1)
-        return self.evaluate_agent_pair(agent_pair, num_games=num_games, display=display)
-
-    def evaluate_optimal_pair(self, num_games, delivery_horizon=2, display=False):
-        a0 = CoupledPlanningAgent(self.mlp, delivery_horizon=delivery_horizon)
-        a1 = CoupledPlanningAgent(self.mlp, delivery_horizon=delivery_horizon)
-        a0.mlp.env = self.env
-        a1.mlp.env = self.env
-        agent_pair = AgentPair(a0, a1)
-        return self.evaluate_agent_pair(agent_pair, num_games=num_games, display=display)
-
-    def evaluate_one_optimal_one_random(self, num_games, display=True):
-        a0 = CoupledPlanningAgent(self.mlp)
-        a1 = RandomAgent()
-        agent_pair = AgentPair(a0, a1)
-        return self.evaluate_agent_pair(agent_pair, num_games=num_games, display=display)
-
-    def evaluate_one_optimal_one_greedy_human(self, num_games, h_idx=0, display=True):
-        h = GreedyHumanModel(self.mlp)
-        r = CoupledPlanningAgent(self.mlp)
-        agent_pair = AgentPair(h, r) if h_idx == 0 else AgentPair(r, h)
-        return self.evaluate_agent_pair(agent_pair, num_games=num_games, display=display)
-
-    def evaluate_agent_pair(self, agent_pair, num_games, game_length=None, start_state_fn=None, metadata_fn=None, metadata_info_fn=None, display=False, info=True):
+        return self.evaluate_agent_pair(agent_pair, num_games=num_games, display=display, regen_mdp=regen_mdp)
+
+    # Depreciated. Need to amend Heuristic before reactivating CoupledPlanningAgent:
```

Comment:
I believe this is a typo: "Deprecated". I think I might have seen this before elsewhere in the code, might be worth to command f in the codebase and replace all instances

Reply:
OK, sounds good.